pkg/plugins/runtime/gateway/route_table_generator.go

```diff
@@ -36,6 +36,16 @@ func GenerateVirtualHost(
 		)
 	}
 
+	routeBuilder := route.RouteBuilder{}
+
+	if len(routes) == 0 {
+		routeBuilder.Configure(route.RouteActionDirectResponse("This is a KumaGateway! No route specified in your gateway"))
```

Comment:
This case should be return a `404` [which keeps it also in line with gateway API](https://github.com/kubernetes-sigs/gateway-api/blob/9537337d838cfebe3775c51be83c486a127c0d88/apis/v1alpha2/httproute_types.go#L166-L167)